compiler/src/lntoamm/UserFunction.ts

```diff
@@ -654,12 +654,50 @@ ${statements[i].statementAst.t.trim()} on line ${statements[i].statementAst.line
     // after this one so we don't mark non-recursive calls to a function multiple times as recursive
     // TODO: This is not the most efficient way to do things, come up with a better metadata
     // mechanism to pass around.
+    let tailed: [Microstatement, Array<Microstatement>] | null = null;
     for (let i = originalStatementLength; i < microstatements.length; i++) {
       if (microstatements[i].statementType === StatementType.ENTERFN) {
         microstatements.splice(i, 1)
         i--
+      } else if (microstatements[i].statementType === StatementType.TAIL) {
+        tailed = [
+          // TAIL mStmt
+          microstatements.splice(i, 1)[0],
+          // everything that goes in the tail function. even if there are no more elements,
+          // assume we still need to provide an empty closure.
```

Comment:
Currently an empty closure causes AGA to AGC to fail, and the AVM has no idea how to handle a handler with no instructions. Is there something that can be done instead in that situation?

Reply:
per an offline conversation, we agreed to create an AVM `nop` closure (cc #433)